x-pack/lib/filters/geoip/database_manager.rb

```diff
@@ -61,6 +63,15 @@ def initialize(geoip, database_path, database_type, vendor_path)
 
   public
 
+  # create data dir, path.data, for geoip if it doesn't exist
+  # copy CC databases to data dir
+  def self.prepare_cc_db
+    FileUtils::mkdir_p(get_data_dir)
+    FileUtils.cp_r(::Dir.glob(::File.join(
+      LogStash::Environment::LOGSTASH_HOME, "vendor", "**", "{GeoLite2-ASN,GeoLite2-City}.mmdb")),
+      get_data_dir) if !::File.exist?(get_file_path(CITY_DB_NAME)) || !::File.exist?(get_file_path(ASN_DB_NAME))
```

Comment:
Probably a more readable version could be with not inlined condition:
```suggestion
   unless ::File.exist?(get_file_path(CITY_DB_NAME)) && ::File.exist?(get_file_path(ASN_DB_NAME))
     db_source_dir = ::Dir.glob(::File.join(
           LogStash::Environment::LOGSTASH_HOME, "vendor", "**", "{GeoLite2-ASN,GeoLite2-City}.mmdb"))
     FileUtils.cp_r(,db_source_dir, get_data_dir)
   end   
```